src/main/java/com/adyen/checkout/api/WebhookResource.java

```diff
@@ -26,42 +29,68 @@ public class WebhookResource {
     public WebhookResource(ApplicationProperty applicationProperty) {
         this.applicationProperty = applicationProperty;
 
-        if(this.applicationProperty.getHmacKey() == null) {
+        if (this.applicationProperty.getHmacKey() == null) {
             log.warn("ADYEN_HMAC_KEY is UNDEFINED (Webhook cannot be authenticated)");
             //throw new RuntimeException("ADYEN_HMAC_KEY is UNDEFINED");
         }
     }
 
     /**
-     * Process incoming Webhook notifications
+     * Process incoming Webhook notification: get NotificationRequestItem, validate HMAC signature,
+     * consume the event asynchronously, send response ["accepted"]
+     *
      * @param notificationRequest
      * @return
      */
     @PostMapping("/webhooks/notifications")
-    public ResponseEntity<String> webhooks(@RequestBody NotificationRequest notificationRequest){
-
-        notificationRequest.getNotificationItems().forEach(
-          item -> {
-              // We recommend validate HMAC signature in the webhooks for security reasons
-              try {
-                  if (new HMACValidator().validateHMAC(item, this.applicationProperty.getHmacKey())) {
-                      log.info("Received webhook with event {} : \n" +
-                          "Merchant Reference: {}\n" +
-                          "Alias : {}\n" +
-                          "PSP reference : {}"
-                          , item.getEventCode(), item.getMerchantReference(), item.getAdditionalData().get("alias"), item.getPspReference());
-                  } else {
-                      // invalid HMAC signature: do not send [accepted] response
-                      log.warn("Could not validate HMAC signature for incoming webhook message: {}", item);
-                      throw new RuntimeException("Invalid HMAC signature");
-                  }
-              } catch (SignatureException e) {
-                  log.error("Error while validating HMAC Key", e);
-              }
-          }
-        );
-
-        // Notifying the server we're accepting the payload
+    public ResponseEntity<String> webhooks(@RequestBody NotificationRequest notificationRequest) {
+
+        // fetch first (and only) NotificationRequestItem
+        var notificationRequestItem = notificationRequest.getNotificationItems().stream().findFirst();
+
+        if (notificationRequestItem.isPresent()) {
+
+            var item = notificationRequestItem.get();
+
+            try {
+                if (new HMACValidator().validateHMAC(item, this.applicationProperty.getHmacKey())) {
```

Comment:
Would it be possible to dependency inject the HMACValidator instance? It seems inefficient to create a new one for every request.